src/server/dfly_bench.cc

```diff
@@ -119,6 +120,91 @@ struct ShardInfo {
 
 using ClusterSpec = vector<ShardInfo>;
 
+template <typename D>
+ClusterSpec FetchCluster(const unique_ptr<FiberSocketBase, D>& socket,
```

Comment:
You don't have to make this a template and passing `unique_ptr` as `const &` is an `anti-pattern`. I suggest to use a plain ptr here. `FibserSocketBase*`